crates/astria-sequencer/src/grpc/sequencer.rs

```diff
@@ -115,18 +94,58 @@ impl SequencerService for SequencerServer {
             rollup_ids.push(rollup_id);
         }
 
-        // XXX: This is a potentially very expensive operation. The the cometbft block
-        // could be pulled async, and the conversion to a sequencer block be performed
-        // in a thread/blocking task.
-        let block = match self.client.sequencer_block(height).await {
-            Ok(block) => block.into_filtered_block(rollup_ids),
-            Err(e) => {
-                return Err(Status::internal(format!(
-                    "failed to get sequencer block from cometbft: {e}",
-                )));
-            }
+        let block_hash = snapshot
+            .get_block_hash_by_height(request.height)
+            .await
+            .map_err(|e| Status::internal(format!("failed to get block hash from storage: {e}")))?;
+
+        let (header, rollup_transactions_root, _) = snapshot
+            .get_sequencer_block_header_by_hash(&block_hash)
+            .await
+            .map_err(|e| {
+                Status::internal(format!(
+                    "failed to get sequencer block header from storage: {e}"
+                ))
+            })?
+            .into_values();
+
+        let (rollup_transactions_proof, rollup_ids_proof) = snapshot
+            .get_block_proofs_by_block_hash(&block_hash)
+            .await
+            .map_err(|e| {
+                Status::internal(format!(
+                    "failed to get sequencer block proofs from storage: {e}"
+                ))
+            })?;
+
+        let all_rollup_ids = snapshot
+            .get_rollup_ids_by_block_hash(&block_hash)
+            .await
+            .map_err(|e| Status::internal(format!("failed to get rollup ids from storage: {e}")))?
+            .into_iter()
+            .map(RollupId::to_vec)
+            .collect::<Vec<_>>();
+
+        let mut rollup_transactions = Vec::with_capacity(rollup_ids.len());
+        for rollup_id in rollup_ids {
+            let rollup_data = snapshot
+                .get_rollup_data(&block_hash, &rollup_id)
+                .await
+                .map_err(|e| {
+                    Status::internal(format!("failed to get rollup data from storage: {e}",))
+                })?;
+            rollup_transactions.push(rollup_data.into_raw());
+        }
+
+        let block = RawFilteredSequencerBlock {
+            header: Some(header.into()),
+            rollup_transactions,
+            rollup_transactions_root: rollup_transactions_root.to_vec(),
+            rollup_transactions_proof: rollup_transactions_proof.into(),
+            rollup_ids_proof: rollup_ids_proof.into(),
+            all_rollup_ids,
```

Comment:
Does `FilteredSequencerBlock` need `all_rollup_ids` or does it only need the rollup_ids root + proof?

Reply:
it'll still need all the rollup IDs for conductor, so conductor can verify if its rollup ID was included or not (and not omitted in the case of an untrusted/malicious sequencer node)